### ISSUE TITLE
NPM based lsp servers fail to install error code NPM: 243

### ISSUE THREAD

**Issue by HarbingerNight** [williamboman/mason.nvim#1452]
### I've searched open issues for similar requests

- [X] Yes

### I've recently downloaded the latest plugin version of mason.nvim

- [X] Yes

### Problem description

When installing npm based language servers the installation fails with NPM error code 243 which seems to be an access denied error on the installation directory.

I have manually done the same steps in the Mason log up to and including the failed step ( `npm install <server>` ):
  `mkdir -p …/staging/<lsp-name>`
  `npm init --yes --scope=mason`
  `npm install <lsp-name>`
everything succeeds w/o issue or access errors. Once reopening nvim get the same mason error.

### Expected behavior

Packages to successfully install.

### Affected packages

typscript-langauge-server, yaml-language-server, json-lsp

### Mason output

```Text
Failed (1)
  ◍ json-lsp
    ▼ Displaying full log                                                                                                            
      Initialized npm root.
      Installing npm package vscode-langservers-extracted@4.7.0…
      spawn: npm failed with exit code 243 and signal 0.
```


### Installation log

```Text
[INFO  Tue 15 Aug 2023 09:39:54 PM MDT] .../share/nvim/lazy/mason/lua/mason-core/installer/init.lua:184: Executing installer for Package(name=json-lsp) {}
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] .../share/nvim/lazy/mason/lua/mason-core/installer/init.lua:58: Attempting to lock package Package(name=json-lsp)
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] .../share/nvim/lazy/mason/lua/mason-core/installer/init.lua:70: Wrote lockfile Package(name=json-lsp)
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] ...calke/.local/share/nvim/lazy/mason/lua/mason-core/fs.lua:71: fs: mkdirp /home/calke/.local/share/nvim/mason/staging/json-lsp
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] ...im/lazy/mason/lua/mason-core/installer/registry/init.lua:148: Compiling installer. json-lsp {}
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] ...im/lazy/mason/lua/mason-core/installer/registry/init.lua:109: Parsing spec json-lsp {}
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] ...vim/lazy/mason/lua/mason-core/installer/managers/npm.lua:28: npm: init
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "init", "--yes", "--scope=mason" },
  cwd = "/home/calke/.local/share/nvim/mason/staging/json-lsp"
}
[DEBUG Tue 15 Aug 2023 09:39:54 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:162: Spawned with pid 1545200
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:148: Job pid=1545200 exited with exit_code=0, signal=0
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "version", "--json" }
}
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:162: Spawned with pid 1545237
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:148: Job pid=1545237 exited with exit_code=0, signal=0
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] ...vim/lazy/mason/lua/mason-core/installer/managers/npm.lua:63: npm: install "vscode-langservers-extracted" "4.7.0" {}
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:116: Spawning cmd="npm", spawn_opts={
  args = { "install", "vscode-langservers-extracted@4.7.0" },
  cwd = "/home/calke/.local/share/nvim/mason/staging/json-lsp"
}
[DEBUG Tue 15 Aug 2023 09:39:55 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:162: Spawned with pid 1545276
[DEBUG Tue 15 Aug 2023 09:39:56 PM MDT] .../.local/share/nvim/lazy/mason/lua/mason-core/process.lua:148: Job pid=1545276 exited with exit_code=243, signal=0
[DEBUG Tue 15 Aug 2023 09:39:56 PM MDT] ...calke/.local/share/nvim/lazy/mason/lua/mason-core/fs.lua:59: fs: unlink /home/calke/.local/share/nvim/mason/staging/json-lsp.lo
ck
[DEBUG Tue 15 Aug 2023 09:39:56 PM MDT] ...calke/.local/share/nvim/lazy/mason/lua/mason-core/fs.lua:47: fs: rmrf /home/calke/.local/share/nvim/mason/staging/json-lsp
[DEBUG Tue 15 Aug 2023 09:39:56 PM MDT] .../share/nvim/lazy/mason/lua/mason-core/installer/init.lua:39: Building receipt for Package(name=json-lsp)
[ERROR Tue 15 Aug 2023 09:39:56 PM MDT] .../share/nvim/lazy/mason/lua/mason-core/installer/init.lua:249: Installation failed for Package(name=json-lsp) error=spawn: npm f
ailed with exit code 243 and signal 0.
```


### Neovim version (>= 0.7)

NVIM v0.9.1
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/cc -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wvla -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -fno-common -Wno-unused-result -Wimplicit-fallthrough -fdiagnostics-color=always -fstack-protector-strong -DUNIT_TESTING -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -I/build/nvim/parts/nvim/build/.deps/usr/include/luajit-2.1 -I/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/build/src/nvim/auto -I/build/nvim/parts/nvim/build/build/include -I/build/nvim/parts/nvim/build/build/cmake.config -I/build/nvim/parts/nvim/build/src -I/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include -I/build/nvim/parts/nvim/build/.deps/usr/include

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "/usr/share/nvim"

Run :checkhealth for more info

### Operating system/version

Linux blackbox 6.2.0-26-generic #26~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Jul 13 16:27:29 UTC 2 x86_64 x86_64 x86_64 GNU/Linux

### Healthcheck

```Text
mason: require("mason.health").check()                                                                                                               
mason.nvim                                                                                                                                           - OK mason.nvim version v1.6.2                                                                                                                       - OK PATH: append
- OK Providers:                                                                                                                                        mason.providers.registry-api
  mason.providers.client                                                                                                                             - OK neovim version >= 0.7.0                                                                                                                                                                                                                                                                              mason.nvim [Registries]                                                                                                                              - OK Registry github.com/mason-org/mason-registry version: 2023-08-16-gentle-hedge is installed.

mason.nvim [Core utils]
- OK unzip: UnZip 6.00 of 20 April 2009, by Debian. Original by Info-ZIP.
- OK wget: GNU Wget 1.21.2 built on linux-gnu.
- ERROR curl: not available
  - ADVICE:
    - spawn: curl failed with exit code - and signal -. curl is not executable
- OK gzip: gzip 1.10                                                                                                                                 - OK tar: tar (GNU tar) 1.34                                                                                                                         - OK bash: GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)
- OK sh: Ok                                                                                                                                                                                                                                                                                               mason.nvim [Languages]                                                                                                                               - WARNING Go: not available                                                                                                                            - ADVICE:                                                                                                                                              - spawn: go failed with exit code - and signal -. go is not executable
- WARNING Composer: not available
  - ADVICE:
    - spawn: composer failed with exit code - and signal -. composer is not executable                                                               - WARNING PHP: not available                                                                                                                           - ADVICE:                                                                                                                                              - spawn: php failed with exit code - and signal -. php is not executable                                                                         - WARNING Ruby: not available                                                                                                                          - ADVICE:
    - spawn: ruby failed with exit code - and signal -. ruby is not executable                                                                       - WARNING RubyGem: not available
  - ADVICE:                                                                                                                                              - spawn: gem failed with exit code - and signal -. gem is not executable
- WARNING javac: not available
  - ADVICE:
    - spawn: javac failed with exit code - and signal -. javac is not executable
- WARNING java: not available                                                                                                                          - ADVICE:
    - spawn: java failed with exit code - and signal -. java is not executable
- WARNING julia: not available
  - ADVICE:
    - spawn: julia failed with exit code - and signal -. julia is not executable
- OK python: Python 3.10.12
- WARNING node: failed to parse version
  - ADVICE:
    - Error: .../calke/.local/share/nvim/lazy/mason/lua/mason/health.lua:242: attempt to index local 'version' (a nil value)
- WARNING cargo: not available
  - ADVICE:
    - spawn: cargo failed with exit code 1 and signal 0.
- OK luarocks: /usr/bin/luarocks 3.8.0
- WARNING npm: failed to parse version
  - ADVICE:
    - Error: .../calke/.local/share/nvim/lazy/mason/lua/mason/health.lua:228: attempt to index local 'version' (a nil value)
- OK pip: pip 22.0.2 from /usr/lib/python3/dist-packages/pip (python 3.10)
- OK python venv: Ok

mason.nvim [GitHub]
- OK GitHub API rate limit. Used: 0. Remaining: 60. Limit: 60. Reset: Tue 15 Aug 2023 11:02:48 PM MDT.
  Install and authenticate via gh-cli to increase rate limit.
```


### Screenshots

_No response_

**Comment by williamboman** [comment]
Hello! The healthcheck suggests Mason might be having issues with `node` and `npm`. Most likely your shell is picking up a different installation than Mason. In Neovim, are you able to check the versions manually?

```
:!node --version
:!npm --version
```

**Comment by HarbingerNight** [comment]
I ran into this issue while trying to solve a different one. There was only on install of node and npm on the system and it was in the path.

The original issue was node based lsp servers would launch but not attach from/to nvim. All other language servers worked. The servers worked when tested w/o nvim. They would launch but not respond back. This same issue occured using the minimum lsp config and when starting the servers through the nvim command line.

This issue turned out to be an issue with the snap distro of node. Would occure with a clean ubuntu 20 install and the snap versions of node and nvim. Once I installed node from somewhere othef than snap the issue went away.
